pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagentplan.go

```diff
@@ -117,25 +125,38 @@ func (h *handler) syncSystemUpgradeControllerStatus(obj *rkev1.RKEControlPlane,
 	// we expect the PSP value to be explicitly defined as either true or false
 	valuesYamlAvailable := sucBundle.Spec.Helm != nil && sucBundle.Spec.Helm.Values != nil
 	if !valuesYamlAvailable {
-		rke2.SystemUpgradeControllerReady.Reason(&status, "Waiting for Upgraded System Upgrade Controller Deployment")
+		rke2.SystemUpgradeControllerReady.Reason(&status, "waiting for system-upgrade-controller bundle values availability")
+		rke2.SystemUpgradeControllerReady.Message(&status, "")
+		rke2.SystemUpgradeControllerReady.Unknown(&status)
 		return status, nil
 	}
 
 	// look through the values yaml content to determine if 'psp: enabled: true'
-	rke2.SystemUpgradeControllerReady.Reason(&status, "Waiting for System Upgrade Controller Bundle Update")
```

Comment:
So was the status thrashing due to this call immediately updating the status of the object? I had the assumption the object's status wouldn't be updated until this function returns, but if thats not then the case then that makes sense

Reply:
The issue stems from the fact that the use of a condition helper function updates the timestamp of the condition in question.

By calling `Unknown` unconditionally, it would cause the status to get unnecessarily updated as the timestamp was changing.

Reply:
Ah okay, so the status is only updated when the function returns but because we went through a state change the timestamp was updated. I'll keep that in mind for the future